include/rpm/rpmpgp.h

```diff
@@ -1113,6 +1119,95 @@ int pgpDigParamsCmp(pgpDigParams p1, pgpDigParams p2);
  */
 unsigned int pgpDigParamsAlgo(pgpDigParams digp, unsigned int algotype);
 
+/** \ingroup rpmpgp
+ * Returns the issuer or the object's Key ID.
+ *
+ * If the object is a signature, then this returns the Key ID stored in the
+ * first Issuer or Issuer Fingerprint subpacket as a hex string.
+ * (This is not authenticated.)
+ *
+ * If the object is a certificate or a subkey, then this returns the key's
+ * Key ID.
+ *
+ * The caller must not free the returned buffer.
+ *
+ * param digp		parameter container
+ * return		an array of PGP_KEYID_LEN bytes.  If the issuer is
+ * 			unknown, this returns an array with all zeros.
+ */
+const uint8_t *pgpDigParamsSignID(pgpDigParams digp);
+
+/** \ingroup rpmpgp
+ * Retrieve the primary User ID, if any.
+ *
+ * Returns the primary User ID, if any.
+ *
+ * If the object is a signature, then this returns NULL.
+ *
+ * If the object is a certificate or a subkey, then this returns the
+ * certificate's primary User ID, if any.
+ *
+ * This interface does not provide a way for the caller to recognize
+ * any embedded NUL characters.
+ *
+ * The caller must not free the returned buffer.
+ *
+ * param digp		parameter container
+ * return		a string or NULL, if there is no primary User ID.
+ */
+const char *pgpDigParamsUserID(pgpDigParams digp);
+
+/** \ingroup rpmpgp
+ * Retrieve the object's version.
+ *
+ * Returns the object's version.
+ *
+ * If the object is a signature, then this returns the version of the
+ * signature packet.
+ *
+ * If the object is a certificate, then this returns the version of the
+ * primary key packet.
+ *
+ * If the object is a subkey, then this returns the version of the subkey's
+ * key packet.
+ *
+ * param digp		parameter container
+ * return		the object's version
+ */
+int pgpDigParamsVersion(pgpDigParams digp);
+
+/** \ingroup rpmpgp
+ * Retrieve the object's creation time.
+ *
+ * If the object is a signature, then this returns the signature's creation
+ * time.
+ *
+ * If the object is a certificate, then this returns the primary key's key
+ * creation time.
+ *
+ * If the object is a subkey, then this returns the subkey's key creation
+ * time.
+ *
+ * param digp		parameter container
+ * return		seconds since the UNIX Epoch.
+ */
+uint32_t pgpDigParamsTime(pgpDigParams digp);
```

Comment:
```suggestion
uint32_t pgpDigParamsCreationTime(pgpDigParams digp);
```

There are other timestamps that an object can have, so I suggest being specific as to which one this is.

Reply:
The tricky part is that this function is already overloaded.  In the case of signatures, the time stamp refers to the [signature creation time](https://datatracker.ietf.org/doc/html/rfc4880#section-5.2.3.4).  In the case of keys, the time stamp refers to the key's [key creation time](https://datatracker.ietf.org/doc/html/rfc4880#section-5.5.2).  Of course any valid key also has a valid binding signature, which also has a signature creation time :/.  But, since they are both creation times, your suggestion is an improvement.